NetApp RPS Azure Sentinel Solution

SoumyaRanjanNaik opened the pull request:
Change(s):
   - Added NetApp RPS Solution

Reason for Change(s):
   - Provides the base capability playbooks for NetApp Ransomware Protection Service integration

Testing Completed:
   - Yes

Checked that the validations are passing and have addressed any issues that are present:
   - Resource ID Issue, help needed

v-maheshbh commented:
Hi @SoumyaRanjanNaik The required properties ‘tactics’, ‘relevantTechniques’, and associated analytical rules are missing from the input data.

Solution Validation – SolutionMetadata:
Validation Error: The provided domain(s) [Security - Ransomware Protection] are invalid or not recognized. Please provide a valid domain. 

BasePath Format:
Modify the BasePath to the following structure:
C:\GitHub\Azure-Sentinel\solutions\Solution Name

offerId Format:
Update the offerId to follow the correct format:
azure-sentinel-solution-solutionName

Please address these items at your earliest convenience. Let me know if you have any questions or need further clarification.

Thanks!

SoumyaRanjanNaik commented:
> Hi @SoumyaRanjanNaik The required properties ‘tactics’, ‘relevantTechniques’, and associated analytical rules are missing from the input data.
> 
> Solution Validation – SolutionMetadata: Validation Error: The provided domain(s) [Security - Ransomware Protection] are invalid or not recognized. Please provide a valid domain.
> 
> BasePath Format: Modify the BasePath to the following structure: C:\GitHub\Azure-Sentinel\solutions\Solution Name
> 
> offerId Format: Update the offerId to follow the correct format: azure-sentinel-solution-solutionName
> 
> Please address these items at your earliest convenience. Let me know if you have any questions or need further clarification.
> 
> Thanks!

Hi @v-maheshbh, have addressed the issues now. Will wait for checks to run. Thanks!

SoumyaRanjanNaik commented:
Hi @v-maheshbh, I have resolved the conflicts and updated the PR. Please review. 

Thanks.

v-maheshbh commented:
Hi @SoumyaRanjanNaik 

Kindly add release notes that include the date, version number, and comment. and update the logo in correct format.
please refer below solution.
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Cloudflare/Data/Solution_Cloudflare.json

Thanks!

v-maheshbh commented:
Hi @SoumyaRanjanNaik 
Kindly review the comments shared above.

Thanks!

v-maheshbh commented:
Hi @SoumyaRanjanNaik 
Kindly package this solution using V3 tool - [https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V3/README.md](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2FAzure-Sentinel%2Fblob%2Fmaster%2FTools%2FCreate-Azure-Sentinel-Solution%2FV3%2FREADME.md&data=05%7C02%7Cv-maheshbh%40microsoft.com%7C4445c6f6310344b203dc08de68801b6d%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C639063096629838693%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=1R%2BioMFIQUOKSiyAcDpM1iLFmy60qrpqm36GClr2Wgk%3D&reserved=0).

Below are the ARM‑TTK validation failures identified during template validation.

<img width="851" height="194" alt="image" src="https://github.com/user-attachments/assets/23ecff3a-dbfc-4d16-b9ad-ae332bda6e99" />

<img width="593" height="188" alt="image" src="https://github.com/user-attachments/assets/da3ca4cd-bdda-4a1d-9234-9cbaa5aec2c6" />

Thanks!

SoumyaRanjanNaik commented:
@v-maheshbh any other details required for getting the PR reviewed and approved?
